### ISSUE TITLE
link to the v2 LESS object for plugin management?

### ISSUE THREAD

**Issue by Pomax** [less/less-docs#330]
Reading through https://github.com/less/less-docs/blob/master/content/usage/plugins.md it seems that the section for [plugin authors](https://github.com/less/less-docs/blob/master/content/usage/plugins.md#for-plugin-authors) is missing a link to the API for "the less object v2" so that plugin authors can look at how they can use this object, rather than having to dig through code other people have already written by inspecting the github repos for the linked plugins.

**Comment by seven-phases-max** [comment]
There's no such page. In other words the only existing documentation for plugin authour is the sources themeselves. For a quick list of useful (for your JSON renderer) references see: https://github.com/postcss/postcss/issues/320#issuecomment-98518577.

**Comment by Pomax** [comment]
In that case please consider this both a request for a link to docs, with an implicit request for docs =)

(for less to json, those links help, but for discovering what capabilities plugins can have is quite a bit of work right now. There is no place to discover that LESS has a postprocessor and a node visitor, for instance, except by reading through the plugin source code)

**Comment by seven-phases-max** [comment]
And I keep it open.

>There is no place to discover that LESS has a postprocessor and a node visitor

quote from http://lesscss.org/usage/#plugins-for-plugin-authors:

>Here are some example repos showing the different plugin types **post-processor**: https://github.com/less/less-plugin-clean-css **visitor**: https://github.com/less/less-plugin-inline-urls file-manager: https://github.com/less/less-plugin-npm-import

**Comment by Pomax** [comment]
Yes, that was my point. The way to discover those two types exist is from the paragraph that points you to example source code, but there is currently no dedicated place where prospective plugin authors can find the full list of supported plugin types, and which bindings they need to expose to do their job. I appreciate that paragraph existing at all, but as someone interested in writing a LESS plugin, it's too tucked away and phrased in such a way to -certainly at first glance- seem footnote information instead of important information to consider and follow up on.

**Comment by matthew-dean** [comment]
I'd love it if someone wrote a Medium article (or Smashing Magazine, whatever's appropriate) on using plugins, especially inline plugins.
